test/python/algorithms/optimizers/test_optimizers.py

```diff
@@ -155,5 +171,215 @@ def callback(x):
         self.assertTrue(values)  # Check the list is nonempty.
 
 
+@ddt
+class TestOptimizerSerialization(QiskitAlgorithmsTestCase):
+    """Tests concerning the serialization of optimizers."""
+
+    @data(
+        ("BFGS", {"maxiter": 100, "eps": np.array([0.1])}),
+        ("CG", {"maxiter": 100, "gtol": 1e-8}),
+        ("COBYLA", {"maxiter": 100}),
```

Comment:
Hopefully maxiter of 100 is not some magic number - would be nice to see some other values as a bit more variety to the test.

Reply:
It shouldn't be 😉 I'll change some